### ISSUE TITLE
[feature request]Changing how the loss metric get the shape information

### ISSUE THREAD

**Issue by labor00** [pytorch/ignite#293]
Recently I was working with multi-output models and I was using a pattern that I believe to
be common. The model returns a tuple, for instance `y_pred = (ypred_1, ypred_2)` with the correspondent y of the form `y = (y_1, y_2)` with the `loss_fn` accepting as arguments `ypred` and  `y` (`loss_fn(ypred, y)`).

However I have run into problems  when using the Loss metric due to the batch size used on the update function of this metric calling the attribute shape of the, presumed, torch.Tensor directly.

I have as suggestion to change the Loss metric class to accept an extra function, for instance `batch_size = lambda x: x.shape[0]` to recover the current behavior and to permit to access the shape attribute from a tuple/list.

**Comment by vfdev-5** [comment]
@labor00 thanks for the feedback! It's true that `Loss` is not working for a generic loss function. Previously, we decided to ask users [to override `Loss` function](https://github.com/pytorch/ignite/pull/112) in such cases. However maybe it's time to rethink the purpose of `Loss` and add such feature... 

cc @jasonkriss

**Comment by labor00** [comment]
Yes I am using  a custom loss.
